pkg/controller/csr/csr_controller.go

```diff
@@ -271,9 +306,31 @@ func (r *reconcileCSR) Reconcile(ctx context.Context, request reconcile.Request)
 // - Verify that the CSR was not previously denied or failed.
 // - Verify that the public key matches the signature on the CSR for the provider algorithm.
 // - Key usages are fixed, so the CSR won't be able to affect these settings.
-func (r *reconcileCSR) validate(csr *certificatesv1.CertificateSigningRequest, pod *corev1.Pod) (*x509.Certificate, error) {
-	if pod == nil {
-		return nil, fmt.Errorf("invalid: no pod can be associated with CSR %s", csr.Name)
+type PodOrHostEndpoint interface {
```

Comment:
The validate func doc string is now detached from the func, due to this interface being in between.

Reply:
Re-attached. The interface definition is moved above the doc string.